api/scpca_portal/utils.py

```diff
@@ -50,3 +56,26 @@ def filter_dict_list_by_keys(list_of_dicts, included_keys, *, ignore_value_error
         new_list_of_dicts.append(dictionary_copy)
 
     return new_list_of_dicts
+
+
+def get_key_superset_from_dicts(list_of_dicts: List[Dict]) -> Set:
+    """Extracts each provided dictionary's key set and returns the superset of all key sets."""
+    key_superset = set()
+    for dictionary in list_of_dicts:
+        key_superset.update(set(dictionary.keys()))
+
+    return key_superset
```

Comment:
I think that we could just have `get_keys_from_dicts` and then the docstring can describe that it is a union of all of the dict's keys.

Also, I don't think `list_of_dicts` is necessary as we have type hinting and we don't generally need to write the type in the variable name.

```suggestion
def get_keys_from_dicts(dicts: List[Dict]) -> Set:
    """Takes a list of dictionaries and returns a set equal to the union of their keys."""
    return set(k for d in dicts for k in d.keys())
```

Reply:
Or if you import `chain` from `itertools` you can do this instead but I think calling `keys` explicitly is a bit nicer
```python
return set(key for key in chain(*dicts))
```

Reply:
Nice. I'll update accordingly.